configs/components/curl.rb

```diff
@@ -1,24 +1,31 @@
 component 'curl' do |pkg, settings, platform|
-  pkg.version '7.51.0'
-  pkg.md5sum '490e19a8ccd1f4a244b50338a0eb9456'
+  pkg.version '7.55.1'
+  pkg.md5sum '3b832160a8c9c40075fd71191960307c'
   pkg.url "https://curl.haxx.se/download/curl-#{pkg.get_version}.tar.gz"
   pkg.mirror "http://buildsources.delivery.puppetlabs.net/curl-#{pkg.get_version}.tar.gz"
 
+  if platform.is_aix?
+    # On AIX, sys/poll.h defines event and revent when _ALL_SOURCE is 1. CURL sets _ALL_SOURCE,
+    # but those collide with definitions in multi.h, causing multi.c to fail because it includes
+    # sys/poll.h (via select.h) and thereby redefines names declared in multi.h.
+    # Patch to disable _ALL_SOURCE when including select.h from multi.c.
```

Comment:
Because there's a longer description in the patch, can you actually just write:

```
# Patch to disable _ALL_SOURCE when including select.h from multi.c. see patch for details.
```

in the comments, instead of the whole reasoning again? I actually prefer shorter comments in the component configs (since the component configs can get very large)

Reply:
I'll update the comment, then merge.